### PR TITLE
WebDriver+Browser: Implement `GET /session/{id}/element/{id}/screenshot`

### PR DESCRIPTION
**Pull request by trflynn89**
[Screencast from 2022-11-04 19-58-53.webm](https://user-images.githubusercontent.com/5600524/200091074-e483068c-a5d3-4ec0-b777-ad3b4b339429.webm)

Ref #15551

### CODE REVIEW

**Comment by trflynn89** [comment]
Hmm might not be totally right - we might be responsible for only saving a screenshot of the part of the element in the viewport? Not sure. Firefox and Chrome behave differently:

Firefox:
1. Window screenshot saves just the visible portion (like us).
2. Element screenshot saves the entire element (like us).

Chrome:
1. Window screenshot saves just the visible portion (like us).
2. Element screenshot saves just the visible portion (*not* like us).

File: Userland/Services/WebDriver/Session.cpp [hunk 1]
```diff
@@ -1354,4 +1354,34 @@ ErrorOr<JsonValue, WebDriverError> Session::take_screenshot()
     return encoded_string;
 }
 
+// 17.1 Take Screenshot, https://w3c.github.io/webdriver/#take-screenshot
```
Comment (linusg) [inline Userland/Services/WebDriver/Session.cpp:1357]:
```suggestion
// 17.2 Take Element Screenshot, https://w3c.github.io/webdriver/#take-element-screenshot
```

**Review by linusg — CHANGES_REQUESTED** [review]

**Comment by linusg** [comment]
It looks like Chrome is right, yeah:

> ...and a [command](https://w3c.github.io/webdriver/#dfn-commands) [Take Element Screenshot](https://w3c.github.io/webdriver/#dfn-take-element-screenshot) for doing the same with the **visible region** of an [element](https://dom.spec.whatwg.org/#concept-element)’s [bounding rectangle](https://w3c.github.io/webdriver/#dfn-bounding-rectangle) after it has been [scrolled into view](https://w3c.github.io/webdriver/#dfn-scrolls-into-view).

Shouldn't this be as simple as intersecting the element rect with the viewport rect?

**Comment by trflynn89** [comment]
> Shouldn't this be as simple as intersecting the element rect with the viewport rect?

It *should* have been except took too long to figure out why that wasn't working. Turns out after scrolling the element into view, there's a series of events that need to happen before the viewport rect is actually updated. That scroll API was async, so it was happening *after* the screenshot was taken.

So I made it sync, and now it works:
[Screencast from 2022-11-04 19-58-53.webm](https://user-images.githubusercontent.com/5600524/200091050-81bb754a-68fc-4a3d-93a4-d1c7c16a7b99.webm)

**Review by linusg — APPROVED** [review]
rad!

**Pull request merged**